docs/architecture/authentication-and-authorization.md

```diff
@@ -0,0 +1,217 @@
+```yaml
+queries:
+    runtimes: "runtime:view"
+    runtime: "runtime:view"
+mutations:
+    createApplication: "application:admin"
+    updateApplication: "application:admin"
+    deleteApplication: "application:admin"
+types:
+    Application:
+        webhooks: "webhook:view"
+        apis: "api:view"
+        eventAPIs: "eventapi:view"
+        documents: "document:view"
+```
+
+The actual scopes will be defined later.
+
+#### Limiting Application/Runtime modifications
+
+Application/Runtime shouldn't be able to modify other Applications or Runtimes. In future, to limit the functionality, we could introduce another GraphQL directive.
+
+```graphql
+type Mutation {
+    updateApplication(id: ID!, in: ApplicationInput!): Application! @secureWithScopes(path: "mutations.updateApplication") @limitModifications(type: APPLICATION, idParamName: "id")
+}
+```
+
+The `limitModifications` mutation would compare ID provided for the `updateApplication` mutation with Application ID saved in the context by Tenant Mapping Handler.
+
+## Authentication flows
+
+Each authentication flow will be handled on a separate host via different VirtualService, as currently OathKeeper doesn't support certificates and multiple `Bearer` authenticators.
+
+### OAuth 2.0 Access Token
+
+**Used by:** Integration System / Application / Runtime
+
+There are two ways of creating a `client_id` and `client_secret` pair in the Hydra, using Hydra's [oauth client](https://github.com/kyma-project/kyma/blob/ab3d8878d013f8cc34c3f549dfa2f50f06502f14/docs/security/03-06-oauth2-server.md#register-an-oauth2-client) or [simple POST request](https://github.com/kyma-incubator/examples/tree/master/ory-hydra/scenarios/client-credentials#setup-an-oauth2-client).
+
+**Obtaining token:**
+
+1. Runtime/Application/IntegrationSystem requests `client_id` and `client_credentials` pair from Director by separate GraphQL mutation. Director generates the pair, registers it in Hydra with proper scopes (defined by object type) and writes it in database.
+1. Runtime/Application/IntegrationSystem calls Hydra with encoded credentials (`client_id` is the ID of `SystemAuth` entry related to given Runtime/Application/IntegrationSystem) and requested scopes.
+1. If the requested scopes are valid, Runtime/Application/IntegrationSystem receives in response an access token, otherwise receives an error.
+
+**Request flow:**
+
+1. Authenticator calls Hydra for introspection of the token.
+1. If the token is valid, OathKeeper sends the request to Hydrator. 
+1. Hydrator calls Tenant Mapping Handler hosted by `Director` to get `tenant` based on a `client_id` (`client_id` is the ID of `SystemAuth` entry related to given Runtime/Application/IntegrationSystem) .
+1. Hydrator passes response to ID_Token mutator which constructs a JWT token with scopes and `tenant` in the payload.
+1. The request is then forwarded to the desired component (such as `Director` or `Connector`) through the `Gateway` component.
```

Comment:
Just a small comment: diagram does not include Connector Service in this flow. IMO we should include it after `Gateway` block as well.